betelgeuse/source_generator.py

```diff
@@ -0,0 +1,347 @@
+"""Generates source code from an AST.
+
+Inpired by the ast.NodeVisitor and unparse.py (see
+https://github.com/python/cpython/blob/master/Tools/parser/unparse.py)
+"""
+import ast
+import io
+import sys
+
+# Large float and imaginary literals get turned into infinities in the AST.  We
+# unparse those infinities to INFSTR.
+INFSTR = '1e' + repr(sys.float_info.max_10_exp + 1)
+
+
+class SourceGenerator():
```

Comment:
Do we need this whole thing? Can't we subclass `Unparser` (or whatever the original class was)?

This is large chunk of code that does stand out and honestly, it's not very clear for me why we need it. From my work on testimony I remember AST already exposes decorators for us, and pytest parameters are always decorators, right? I have not dig much into parameters decorators, but does AST miss something important by default?